soccer/planning/RRTPlanner.cpp

```diff
@@ -17,10 +19,60 @@ using namespace Eigen;
 using namespace Geometry2d;
 
 namespace Planning {
+REGISTER_CONFIGURABLE(RRTPlanner);
+
+ConfigDouble* RRTPlanner::_partialReplanLeadTime;
+
+void RRTPlanner::createConfiguration(Configuration* cfg) {
+    _partialReplanLeadTime = new ConfigDouble(
+        cfg, "RRTPlanner/partialReplanLeadTime", 0.2, "partialReplanLeadTime");
+}
 
 RRTPlanner::RRTPlanner(int maxIterations)
     : _maxIterations(maxIterations), SingleRobotPathPlanner(true) {}
 
+bool shouldFullReplan(float timeIntoPath, MotionInstant current,
+                      MotionInstant goal,
+                      const MotionConstraints& motionConstraints,
+                      const Geometry2d::ShapeSet& obstacles,
+                      const Path* prevPath) {
+    boost::optional<RobotInstant> optTarget = prevPath->evaluate(timeIntoPath);
+    // If we went off the end of the path, use the end for calculations.
+    MotionInstant targetInstant =
+        optTarget ? optTarget->motion : prevPath->end().motion;
+    float pathError = current.pos.distTo(targetInstant.pos);
+    float replanThreshold = (float)*motionConstraints._replan_threshold;
+
+    if (*motionConstraints._replan_threshold != 0 &&
+        pathError > replanThreshold) {
+        return true;
+    }
+
+    return false;
+}
+
+boost::optional<float> shouldPartialReplan(
```

Comment:
Could this method be renamed for clarity?  The word "should" usually means it's a method that returns a boolean.